Mathlib/LinearAlgebra/FreeModule/Finite/CardQuotient.lean

```diff
@@ -93,11 +93,30 @@ end Submodule
 
 section AddSubgroup
 
-theorem AddSubgroup.natAbs_det_basis_change {E : Type*} [AddCommGroup E] {ι : Type*}
+theorem AddSubgroup.index_eq_natAbs_det {E : Type*} [AddCommGroup E] {ι : Type*}
     [DecidableEq ι] [Fintype ι] (bE : Basis ι ℤ E) (N : AddSubgroup E) (bN : Basis ι ℤ N) :
-    (bE.det (bN ·)).natAbs = N.index :=
+    N.index = (bE.det (bN ·)).natAbs :=
   have : Module.Free ℤ E := Module.Free.of_basis bE
   have : Module.Finite ℤ E := Module.Finite.of_basis bE
-  Submodule.natAbs_det_basis_change bE N.toIntSubmodule bN
+  (Submodule.natAbs_det_basis_change bE N.toIntSubmodule bN).symm
+
+theorem AddSubgroup.relindex_eq_natAbs_det {E : Type*} [AddCommGroup E]
+    (L₁ L₂ : AddSubgroup E) (H : L₁ ≤ L₂) {ι : Type*} [DecidableEq ι] [Fintype ι]
+    (b₁ : Basis ι ℤ L₁.toIntSubmodule) (b₂ : Basis ι ℤ L₂.toIntSubmodule) :
+    L₁.relindex L₂ = (b₂.det (fun i ↦ ⟨b₁ i, (H (SetLike.coe_mem _))⟩)).natAbs := by
+  rw [relindex, index_eq_natAbs_det b₂ _ (b₁.map (addSubgroupOfEquivOfLe H).toIntLinearEquiv.symm)]
+  rfl
```

Comment:
Is it possible to avoid this `rfl`?

Reply:
This involves commuting `Int.cast` with a bunch of functions and it's a bit cumbersome.

Reply:
I don't know the good way to that and we don't seem to have the suitable lemmas.